v3/generate.go

```diff
@@ -144,6 +151,51 @@ func duplicateMap(m map[interface{}]interface{}) map[interface{}]interface{} {
 	return cp
 }
 
+// generateEndpointDetails is a helper function that adds endpoint details to, and removes
+// irrelevant endpoints from, the provided paths map
+func generateEndpointDetails(paths map[interface{}]interface{}, shouldDeletePath func(interface{}) bool) {
+	for path, methodOp := range paths {
+		if shouldDeletePath(path) {
+			delete(paths, path)
+			continue
+		}
+
+		for _, o := range methodOp.(map[interface{}]interface{}) {
+			operation := o.(map[interface{}]interface{})
+			params, ok := operation["parameters"].([]interface{})
+			if !ok {
+				continue
+			}
+			paramsForClient := make([]map[interface{}]interface{}, 0)
+			for _, p := range params {
+				param := p.(map[interface{}]interface{})
+				paramsForClient = append(paramsForClient, param)
+			}
+			operation["parameters"] = paramsForClient
```

Comment:
question: wait, so what does this transformation do, exactly?

Reply:
seems like it changes endpoint parameters from a `[]interface{}` to `[]map[interface{}]interface{}` which I assume affects how the file is written / marshalled via the `yaml` [package](https://github.com/Clever/yaml)

although I might want to do a bit of poking for a cleaner way to do it

Reply:
yeah... that does make sense, but I think it would be helpful to document somehow why that is? anyway, I don't think you have to do that here, but the why of this loop is pretty opaque from just reading it

Reply:
yeah I can add a comment for now

Reply:
actually I just tried running the generate code without this block and it looks the same 😂 I bet this was originally ported over from the [v2 generate code](https://github.com/Clever/swagger-api/blob/master/v2/generate.go#L186) but without whatever internal logic that actually made some modifications. going to just entirely remove this block